### PR TITLE
(TK-337) Fix race in timeout test

### PR DESCRIPTION
**Pull request by MSLilah**
Previously, our status callback timeout test had a race condition
that caused it to almost always fail in Travis. This commit
fixes that race by having the status function block indefinitely
rather than sleep for 2 seconds.

### CODE REVIEW

**Comment by MSLilah** [comment]
I need to do a few travis runs to ensure that the race condition is gone

**Comment by MSLilah** [comment]
This should be ready for merge now

File: test/puppetlabs/trapperkeeper/services/status/status_core_test.clj [hunk 1]
```diff
@@ -73,13 +73,15 @@
             (is (= :unknown (:state result))))))
 
       (testing "and it is from a timeout"
-        (update-status-context status-fns "quux" "1.1.0" 1 (fn [_] (Thread/sleep 2000) {:state :running
-                                                                                        :status "aw yis"}))
+        ; Add a status function that will block indefinitely to ensure we will always
+        ; timeout
+        (update-status-context status-fns "quux" "1.1.0" 1 (fn [_] (deref (promise)) {:state :running
```
Comment (KevinCorcoran) [inline test/puppetlabs/trapperkeeper/services/status/status_core_test.clj:78]: I didn't think this change from `(Thread/sleep 2000)` to `(deref (promise))` would be necessary.

Comment (KevinCorcoran) [inline test/puppetlabs/trapperkeeper/services/status/status_core_test.clj:78]: Although I guess it's probably an improvement regardless.

File: test/puppetlabs/trapperkeeper/services/status/status_core_test.clj [hunk 2]
```diff
@@ -73,13 +73,15 @@
...
+                                                                                      :status "aw yis"}))
         (with-redefs [puppetlabs.trapperkeeper.services.status.status-core/check-timeout (constantly 1)]
           (with-test-logging
             (let [result (call-status-fn-for-service "quux" (get @status-fns "quux") :debug)]
               (is (logged? #"Status callback timed out" :error))
-              (is (logged? #"Status callback interrupted" :error))
+              (is (logged? #"CancellationException"))
```
Comment (KevinCorcoran) [inline test/puppetlabs/trapperkeeper/services/status/status_core_test.clj:84]: I'd probably update this PR + commit message with an explanation of why this change was necessary.

File: test/puppetlabs/trapperkeeper/services/status/status_core_test.clj (hunk 1 above)
Comment (MSLilah) [inline test/puppetlabs/trapperkeeper/services/status/status_core_test.clj:78]: Yeah, I figured it was safer to use the `deref`.

**Comment by MSLilah** [comment]
@KevinCorcoran I've updated the commit message

**Comment by KevinCorcoran** [comment]
:+1:

**Comment by jpinsonault** [comment]
:+1:
